python_modules/dagster/dagster/components/component/template_vars.py

```diff
@@ -0,0 +1,47 @@
+import inspect
+from typing import Any, Callable, Optional, Union, overload
+
+from typing_extensions import TypeAlias
+
+from dagster.components.core.context import ComponentLoadContext
+
+TemplateVarFn: TypeAlias = Callable[[ComponentLoadContext], Any]
+
+TEMPLATE_VAR_ATTR = "__dagster_template_var"
+
+
+@overload
+def template_var(fn: TemplateVarFn) -> TemplateVarFn: ...
+
+
+@overload
+def template_var() -> Callable[[TemplateVarFn], TemplateVarFn]: ...
+
+
+def template_var(
```

Comment:
template "var" feels odd since they are functions

* `template_scope`
* `in_template_scope`
* ... 

not strong against `template_var` but worth considering alternatives imo

Reply:
What would the module name that gets loaded be by default? It was `scope.py`. Now it is `template_vars.py`

Reply:
I misread the setup and thought it was just making the functions available not evaluating them and making the result available so this feedback is moot.